independent-projects/bootstrap/app-model/src/main/java/io/quarkus/bootstrap/model/ApplicationModel.java

```diff
@@ -40,6 +40,14 @@ public interface ApplicationModel {
      */
     Iterable<ResolvedDependency> getDependencies(int flags);
 
+    /**
+     * Returns application dependencies that have any of the flags passed in as arguments set.
+     *
+     * @param flags dependency flags to match
+     * @return application dependencies that matched the flags
+     */
+    Iterable<ResolvedDependency> getDependenciesWithAnyFlag(int... flags);
```

Comment:
@ia3andy this is the method you would call passing in `DependencyFlags.RUNTIME_CP, DependencyFlags.COMPILE_ONLY`.